pkg/cloud/packet/client.go

```diff
@@ -180,6 +180,26 @@ func (p *Client) NewDevice(ctx context.Context, req CreateDeviceRequest) (*metal
 
 	serverCreateOpts := metal.CreateDeviceRequest{}
 
+	var sshKeyInput []metal.SSHKeyInput
+	if len(packetMachineSpec.SSHKeys) > 0 {
```

Comment:
We can leave out the `if` conditional. The for loop will only append sshKeyInput if SSHKeys exist (and are >0).
Likewise, if the SSHKeys are nil or 0, sshKeyInput will be nil when sent to the Request object.